advmng.1
.\" Process this file with
.\" groff -man -Tascii advmng.1
.\"
.TH AdvanceMNG 1 "01 Agosto 2002"
.SH NAME
advmng \- mng compression utility
.SH SYNOPSIS
.B advzip [-l, --list] [-z, --recompress]
.B [-0, --shrink-0] [-1, --shrink-1] [-2, --shrink-2] [-3, --shrink-3]
.B [-s, --scroll HxV] [-e, --expand] [-r, --reduce]
.B [-c, --lc] [-C, --vlc] [-f, --force]
.B [-q, --quiet] [-h, --help] [-V, --version]
ARCHIVES... [FILES...]
.SH DESCRIPTION
The main purpose of this utility is to recompress MNG files
to get the smallest possible size.
.PP
It's able to read the MNG file generated by AdvanceMAME and to compress
them using the Delta and the Move MNG features.
Please note that this utility is not able to read a generic MNG file.
.PP
The delta optimization is always active and it can compress video clips
with only small changes between frames.
.PP
The scroll optimization is used only on request (option "-s HxV") and it can
compress better video clips with a scrolling background.
.SH OPTIONS
.B -l, --list ARCHIVES...
List the content of the specified files.
.TP
.B -z, --recompress ARCHIVES...
Recompress the specified archives.
If the -1, -2, -3 options are specified is used the smallest file choice from the
previous compressed data and the new compression.
If the -0 option is specified the archive is always rewritten without any compression.
.TP
.B -0, --shrink-0
Set the compression level to "none". The file are only stored and not compressed.
The file is always rewritten also if it's bigger.
.TP
.B -1, --shrink-1
Set the compression level to "normal". This is the default level of compression.
.TP
.B -2, --shrink-2
Set the compression level to "extra". It's SLOW.
.TP
.B -3, --shrink-3
Set the compression level to "extreme". It's VERY SLOW.
.TP
.B -s, --scroll HxV
The "-s HxV" option specifies the size of the pattern (H width x V height) used to
check for a scrolling background. The size is expressed in pixel and it must
be bigger or equal than the scrolling speed of the background.
For example take the game 1942 that scroll vertically by 1 pixel every frame.
If you recorded with an interleave factor of 2 the vertical scrolling speed
is 1*2 = 2. In this case the minimun pattern is "-s 0x2".
Generally you can use "-s 4x4" and use bigger value only for games scrolling
faster. If the game scroll only horizontally or vertically you can speed up a
lot the compression with monodirectional patterns like "-s 8x0" or "-s 0x8".
.TP
.B -r, --reduce
Force always the color reduction to 8 bit. The compression
will fail if exists a frame with more than 256 colors.
.TP
.B -e, --expand
Force always the color expansion to 24 bit.
.TP
.B -c, --lc
Force the use of the MNG LC (Low Complexity) specifications. It disables the
delta compression.
The file is always rewritten also if it's bigger.
.TP
.B -C, --vlc
Force the use of the MNG VLC (Very Low Complexity) specifications. It disables the
delta compression and the precise timing. Warning! you may lose the correct
timing information because the VLC format only support integer frequency.
The file is always rewritten also if it's bigger.
.TP
.B -f, --force
Force the use of the new file also if it's bigger.
.SH COPYRIGHT
Copyright (C) 2002 Andrea Mazzoleni



HISTORY
AdvanceSCAN Version 1.2 2002/08
	) Added the advpng to compress the PNG files.
	) Added the advmng to compress the MNG files.
	) Changed the output format of the advzip -l option.
		Now it's more like unzip -l.
	) Added the noautomake.sh and autogen.sh scripts.

AdvanceSCAN Version 1.1 2002/06
	) Fixed an infinite loop bug testing some small damaged zips.
	) Removed some warning compiling with gcc 3.1.

AdvanceSCAN Version 1.0 2002/05
	) First public release.
	) Fixed the compression percentage computation on big files.
	) The unknow .zip files are now deleted if they contains only
		empty directories.
	) Added the --pedantic option at the advzip utility. These
		tests are only done if requested.
	) Renamed all the "unknow" options in "unknown".
	) Fixed the reading of configuration file containing the
		\r char.
	) Documentation fixes.
	) Other minor fixes.

AdvanceSCAN Version 0.6-beta 2002/05
	) Major revision.
	) Renamed AdvanceSCAN.
	) Updated to the last C++ standard.
	) General cleanup of the code.
	) Added the AdvanceDIFF and AdvanceZIP utility.

MAMESCAN Version 0.5 2001/08
	) Removed the options -fix, -cat, -collision.

MAMESCAN Version 0.4 2000/02
	) Added configure for msdos.

MAMESCAN Version 0.3

MAMESCAN Version 0.2

MAMESCAN Version 0.1 1999/07
	) First version.

MAMESCAN Version 0.0 1998/11
	) Posted in the MAME list the new listinfo command.



doc/advmng.d
Name
	advmng - The AdvanceMNG compressor

Synopsis
	:advmng [-l, --list] [-z, --recompress] [-0, --shrink-0]
	:[-1, --shrink-1] [-2, --shrink-2] [-3, --shrink-3]
	:[-s, --scroll HxV] [-e, --expand] [-r, --reduce]
	:[-c, --lc] [-C, --vlc] [-f, --force] [-q, --quiet]
	:[-h, --help] [-V, --version] FILES...

Description
	The main purpose of this utility is to recompress MNG
	files to get the smallest possible size.

	It's able to read the MNG files generated by AdvanceMAME
	and to compress them using Delta and Move MNG features.
	Please note that this utility is not able to read
	a generic MNG file.

	The delta optimization is always active and it can
	compress video clips with only small changes between frames.

	The scroll optimization is used only on request (option
	"-s HxV") and it can bettercompress better video clips with
	a scrolling background.

Options
	-l, --list FILES...
		List the content of the specified files.

	-z, --recompress FILES...
		Recompress the specified files. If the -1, -2, -3
		options are specified, it's used the smallest file
		choice from the previous compressed data and the
		new compression. If the -0 option is specified the
		file is always rewritten without any compression.

	-0, --shrink-0
		Set the compression level to "none". The file is
		only stored and not compressed. The file is always
		rewritten also if it's bigger.

	-1, --shrink-1
		Set the compression level to "normal". This is the
		default level of compression.

	-2, --shrink-2
		Set the compression level to "extra". It's SLOW.

	-3, --shrink-3
		Set the compression level to "extreme". It's VERY
		SLOW.

	-s, --scroll HxV
		The "-s HxV" option specifies the size of the pattern
		(H width x V height) used to check for a
		scrolling background. The size is expressed in
		pixels and it must be bigger or equal than the
		scrolling speed of the background. For example
		take the game 1942 that scroll vertically by 1
		pixel every frame. If you recorded with an interleave
		factor of 2 the vertical scrolling speed is
		1*2 = 2. In this case the minimun pattern is "-s 0x2".
		Generally you can use "-s 4x4" and use bigger
		values only for games scrolling faster. If the
		game scrolls only horizontally or vertically you can
		speed up a lot the compression with monodirectional
		patterns like "-s 8x0" or "-s 0x8".

	-r, --reduce
		Force always the color reduction to 8 bit. The
		compression will fail if exists a frame with more than
		256 colors.

	-e, --expand
		Force always the color expansion to 24 bit.

	-c, --lc
		Force the use of the MNG LC (Low Complexity)
		specifications. It disables the delta compression. The
		file is always rewritten also if it's bigger.

	-C, --vlc
		Force the use of the MNG VLC (Very Low Complexity)
		specifications. It disables the delta compression
		and the precise timing. Warning! you may lose the
		correct timing information because the VLC format
		only support integer frequency. The file is always
		rewritten also if it's bigger.

	-f, --force
		Force the use of the new file also if it's bigger.

Bugs
	All the ancillary chuncks are ignored and silently removed.

Copyright
	This file is Copyright (C) 2002 Andrea Mazzoleni, Filipe Estima

See Also
	advzip(1), advpng(1)
